lightblue/src/main/java/org/esbtools/eventhandler/lightblue/LightblueDocumentEventRepository.java

```diff
@@ -103,7 +103,28 @@ public void addNewDocumentEvents(Collection<? extends DocumentEvent> documentEve
                 .map(LightblueDocumentEventRepository::asEntity)
                 .collect(Collectors.toList());
 
-        lightblue.data(InsertRequests.documentEventsReturningOnlyIds(documentEventEntities));
+        int newEventsCount = documentEventEntities.size();
+        int maxEventsPerInsert = config.getMaxDocumentEventsPerInsert().orElse(newEventsCount);
+
+        int numberOfRequestsOfMaxEvents = newEventsCount / maxEventsPerInsert;
```

Comment:
If I understand what you are doing, I think you just want some form of Math.ceiling(newEventsCount / maxEventsPerInsert); and you wouldnt need lines 110 through 114. Might have to do some type casting.